File: src/pl/pe_0002.pl
$gross = 0;
$a = 1;
$b = 2;

while ($a < 4e6) {
    if ($a % 2 == 0) {
        $gross += $a;
    }
    $c = $a + $b;
    $a = $b;
    $b = $c;
}

print $gross . "\n";

File: src/r/pe_0002.r
fibonacci_step <- function(a, b) {
    list(b, a + b)
}

gross <- 0
lst <- list(1, 2)
while (lst[[1]] < 4e6) {
    if (lst[[1]] %% 2 == 0) {
        gross <- gross + lst[[1]]
    }
    lst <- do.call(fibonacci_step, lst)
}

cat(gross, "\n")
